code_generator/GdkCodeGenerator/Templates/UnityComponentConversionGenerator.tt

```diff
@@ -9,493 +9,333 @@
+            public override uint ComponentId => <#= componentDetails.ComponentId #>;
 
-            public override ComponentType[] ReplicationComponentTypes => replicationComponentTypes;
-            private static readonly ComponentType[] replicationComponentTypes = { typeof(<#= componentDetails.TypeName #>), typeof(Authoritative<<#= componentDetails.TypeName #>>), typeof(SpatialEntityId)};
+            private readonly EntityManager entityManager;
 
-            public override ComponentType[] CleanUpComponentTypes => cleanUpComponentTypes;
-            private static readonly ComponentType[] cleanUpComponentTypes = 
-            { 
-                typeof(AuthoritiesChanged<<#= componentDetails.TypeName #>>),
-                typeof(ComponentAdded<<#= componentDetails.TypeName #>>),
-                typeof(ComponentRemoved<<#= componentDetails.TypeName #>>),
-<# if (fieldDetailsList.Count > 0) { #>
-                typeof(ComponentsUpdated<<#= componentDetails.TypeName #>.Update>), 
-<# } #>
-<# foreach(var commandDetails in commandDetailsList) { #>
-                typeof(CommandRequests<<#= commandDetails.CommandName #>.Request>), typeof(CommandResponses<<#= commandDetails.CommandName #>.Response>),
-<# } #>
-<# foreach(var eventDetails in eventDetailsList) { #>
-                typeof(EventsReceived<<#= eventDetails.EventName #>Event>),
-<# } #>
-            };
-<# if(commandDetailsList.Count > 0) { #>
-
-            private readonly Dictionary<uint, RequestContext> RequestIdToRequestContext = new Dictionary<uint, RequestContext>();
-
-<# foreach(var commandDetails in commandDetailsList) { #>
-            internal List<<#= commandDetails.CommandName #>.OutgoingRequest> <#= commandDetails.CommandName #>Requests = new List<<#= commandDetails.CommandName #>.OutgoingRequest>();
-            internal List<<#= commandDetails.CommandName #>.OutgoingResponse> <#= commandDetails.CommandName #>Responses = new List<<#= commandDetails.CommandName #>.OutgoingResponse>();
-            private static readonly ComponentPool<CommandRequests<<#= commandDetails.CommandName #>.Request>> <#= commandDetails.CommandName #>RequestPool =
-                new ComponentPool<CommandRequests<<#= commandDetails.CommandName #>.Request>>(
-                    () => new CommandRequests<<#= commandDetails.CommandName #>.Request>(),
-                    (component) => component.Buffer.Clear());
-            private static readonly ComponentPool<CommandResponses<<#= commandDetails.CommandName #>.Response>> <#= commandDetails.CommandName #>ResponsePool =
-                new ComponentPool<CommandResponses<<#= commandDetails.CommandName #>.Response>>(
-                    () => new CommandResponses<<#= commandDetails.CommandName #>.Response>(),
-                    (component) => component.Buffer.Clear());
-<# } #>
-<# } #>
+            private readonly Dictionary<uint, Action<CommandRequestOp>> CommandRequests;
```

Comment:
Could also just generate these directly, as we know what all the commands are. If keeping this then we should rename it to make it clear what the key is.